code/game/objects/items/toys.dm

```diff
@@ -529,14 +529,10 @@
 
 /obj/item/toy/talking/attack_self(mob/user)
 	if(!cooldown)
-		var/list/messages = generate_messages()
 		activation_message(user)
 		playsound(loc, 'sound/machines/click.ogg', 20, 1)
 
-		spawn(0)
-			for(var/message in messages)
-				toy_talk(user, message)
-				sleep(10)
+		INVOKE_ASYNC(src, .proc/do_toy_talk, user)
```

Comment:
Instead of this, you could instead just add a bunch of timers. Something like
```
var/list/messages = generate_messages()
for(var/i in 1 to LAZYLEN(messages))
	addtimer(CALLBACK(src, .proc/toy_talk, user, messages[i]), (i - 1) * 10)
```
Could be slightly prettier by not creating a timer with length 0 but you get the general point

Reply:
It looks like only `/obj/item/toy/talking/codex_gigas/generate_messages()` generates a list with more than one message, the rest all just use some variation of `list(pick(messages))`
Should I still make this change?